src/vs/vscode.proposed.d.ts

```diff
@@ -168,14 +168,19 @@ declare module 'vscode' {
 		availableFormats: ColorFormat[];
 
 		/**
+		 * Controls whether the color decorator is rendered.
+		 */
+		renderDecorator: boolean;
```

Comment:
I'd go with a simple user setting for now, which can easily be language scoped, and wait for users requesting the very specific _I want color boxes here but not there_ request. We can always add that later.